Base/EventNotification/StateTransition.cs

```diff
@@ -0,0 +1,77 @@
+using System.IO.BACnet.EventNotification.EventValues;
+
+namespace System.IO.BACnet.EventNotification
+{
+    public class StateTransition : EventNotificationData
+    {
+        private readonly EventNotificationData _eventData;
+
+        #region EventNotificationData properties
+
+        public override uint ProcessIdentifier
+        {
+            get => _eventData.ProcessIdentifier;
+            set => _eventData.ProcessIdentifier = value;
+        }
+
+        public override BacnetObjectId InitiatingObjectIdentifier
+        {
+            get => _eventData.InitiatingObjectIdentifier;
+            set => _eventData.InitiatingObjectIdentifier = value;
+        }
+
+        public override BacnetObjectId EventObjectIdentifier
+        {
+            get => _eventData.EventObjectIdentifier;
+            set => _eventData.EventObjectIdentifier = value;
+        }
+
+        public override BacnetGenericTime TimeStamp
+        {
+            get => _eventData.TimeStamp;
+            set => _eventData.TimeStamp = value;
+        }
+
+        public override uint NotificationClass
+        {
+            get => _eventData.NotificationClass;
+            set => _eventData.NotificationClass = value;
+        }
+
+        public override byte Priority
+        {
+            get => _eventData.Priority;
+            set => _eventData.Priority = value;
+        }
+
+        public override string MessageText
+        {
+            get => _eventData.MessageText;
+            set => _eventData.MessageText = value;
+        }
+
+        public override BacnetNotifyTypes NotifyType
+        {
+            get => _eventData.NotifyType;
+            set => _eventData.NotifyType = value;
+        }
+
+        public override BacnetEventStates ToState
+        {
+            get => _eventData.ToState;
+            set => _eventData.ToState = value;
+        }
+
+        #endregion
+
+        public BacnetEventTypes EventType;
+        public bool AckRequired;
+        public BacnetEventStates FromState;
+        public EventValuesBase EventValues;
```

Comment:
I think this is the most important thing in this PR to discuss. We could do it outside the review as it can extend to many posts 😉 So, I don't like two things about this proposal (of course it's way better from what we have now):

1. All the overrides
2. User will need to do a switch (or something similar) on EventValues to see what type of notification he got, cast it (easier in C# 7.0 but still) and then finally he can get the values.

So what I would like, from C# point of view, is to have separete events telling me what kind of notificaiton I received (there can still be one for all of them for backward compatibility, kinda). This way I can get strongly typed 'event values' and no need for switching/casting.

Reply:
1) what would you suggest then? Provide a constructor that copies values from the base type? I've done this, but it came back to bite me. Can't think of anything else at the moment :man_shrugging: 

2) sure, we can do that. I think I'd go with a generic StateTransition Type then - e.g. StateTransition<OutOfRange> - no?

Reply:
1. My suggestion: https://github.com/DarkStarDS9/BACnet/pull/1
2. Yes, I included it in my suggestion